cpp/src/arrow/util/key_value_metadata.h

```diff
@@ -34,11 +34,14 @@ namespace arrow {
 class ARROW_EXPORT KeyValueMetadata {
  public:
   KeyValueMetadata();
-  KeyValueMetadata(const std::vector<std::string>& keys,
-                   const std::vector<std::string>& values);
+  KeyValueMetadata(std::vector<std::string> keys, std::vector<std::string> values);
   explicit KeyValueMetadata(const std::unordered_map<std::string, std::string>& map);
   virtual ~KeyValueMetadata() = default;
 
+  /// \brief Convenience constructor
+  static std::shared_ptr<KeyValueMetadata> Make(std::vector<std::string> keys,
```

Comment:
There's already the `key_value_metadata` convenience function. Is this one really required?

Reply:
Good point. I'll remove this